File: tests/practice_run_tests/java/solution_files/Sol20.java
package tests.practice_run_tests.java.solution_files;
import java.util.Set;
import java.util.ArrayList;
import java.util.Collections;

public class Sol20 {
  public static TreeNode solve(Set<int[]> input) {
    ArrayList<Integer> allVals = new ArrayList<>();
    for (int[] currVals : input) {
      for (int val : currVals) {
        allVals.add(val);
      }
    }

    Collections.sort(allVals);
    return new TreeNode(allVals.get(0), 
                new TreeNode(allVals.get(1), 
                  new TreeNode(allVals.get(2), null, null), 
                  new TreeNode(allVals.get(3), null, null)), 
                new TreeNode(allVals.get(4), null, null));
  }

  public static class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode(int val, TreeNode left, TreeNode right) {
      this.val = val;
      this.left = left;
      this.right = right;
    }
  }
}
  
  

File: tests/practice_run_tests/python/solution_files/sol20.py
from __future__ import annotations 
from typing import Optional

class TreeNode:
  def __init__(self, val, left: Optional[TreeNode] = None, 
               right: Optional[TreeNode] = None):
    self.val = val
    self.left = left
    self.right = right

class Solution:
  def solve(self, input: set[tuple]) -> Optional[TreeNode]:
    all_vals = []
    for tup in input:
      all_vals.extend(tup)
    all_vals.sort()
    if not all_vals:
      return None

    return TreeNode(all_vals[0], 
              TreeNode(all_vals[1], 
                TreeNode(all_vals[2]), 
                TreeNode(all_vals[3])), 
              TreeNode(all_vals[-1]))
  